src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_LockStatement.cs

```diff
@@ -37,6 +40,41 @@ public override BoundNode VisitLockStatement(BoundLockStatement node)
                     argumentType); //need to have a non-null type here for TempHelpers.StoreToTemp.
             }
 
+            if (argumentType.IsWellKnownTypeLock() &&
+                TryGetWellKnownTypeMember(lockSyntax, WellKnownMember.System_Threading_Lock__EnterLockScope, out MethodSymbol enterLockScope) &&
+                TryGetWellKnownTypeMember(lockSyntax, WellKnownMember.System_Threading_Lock__Scope__Dispose, out MethodSymbol lockScopeDispose))
+            {
+                // lock (x) { body } -> using (x.EnterLockScope()) { body }
```

Comment:
>using (x.EnterLockScope()) { body } [](http://example.com/codeflow?start=40&length=35)

This doesn't look like a lowered form, local rewriter is supposed to produce a lowered form, therefore, I would expect the comment to use the lowered form

Reply:
> I would expect the comment to use the lowered form

I think this is fine after all, since we are delegating to ```RewriteUsingStatementTryFinally```.